crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3745,22 +3774,54 @@ impl<'db> TypeInferenceBuilder<'db> {
             ExprContext::Store => {
                 let value_ty = self.infer_expression(value);
 
-                let symbol = if let Type::Instance(instance) = value_ty {
-                    let instance_member = instance.class.instance_member(self.db(), attr);
-                    if instance_member.is_class_var() {
-                        self.context.report_lint(
+                let symbol = match value_ty {
+                    Type::Instance(instance) => {
+                        let instance_member = instance.class.instance_member(self.db(), attr);
+                        if instance_member.is_class_var() {
+                            self.context.report_lint(
                             &INVALID_ATTRIBUTE_ACCESS,
                             attribute,
                             format_args!(
                                 "Cannot assign to ClassVar `{attr}` from an instance of type `{ty}`",
                                 ty = value_ty.display(self.db()),
                             ),
                         );
```

Comment:
some underindentation here

```suggestion
                                &INVALID_ATTRIBUTE_ACCESS,
                                attribute,
                                format_args!(
                                    "Cannot assign to ClassVar `{attr}` from an instance of type `{ty}`",
                                    ty = value_ty.display(self.db()),
                                ),
                            );
```